web-common/src/features/dashboards/time-controls/TimeRangeSelector.svelte

```diff
@@ -99,195 +108,196 @@
       },
       toggleFloatingElement,
     );
+
     dispatch("remove-scrub");
   }
 
-  function onClickOutside(closeMenu: () => void) {
-    if (!isCalendarRecentlyClosed) {
-      closeMenu();
+  function handleMenuOpen() {
+    if (intermediateSelection !== TimeRangePreset.CUSTOM) {
+      isCustomRangeOpen = false;
     }
   }
 
-  // If the user clicks outside to close the calendar, we don't want the `click-outside` event to close the whole menu.
-  // A nice solution would be to check for `isCalendarOpen` in the `onClickOutside` function. However, the calendar
-  // closes *before* the `click-outside` event is fired. A workaround is to check for `isCalendarRecentlyClosed`.
-  function onCalendarClose() {
-    isCalendarRecentlyClosed = true;
-
-    setTimeout(() => {
-      isCalendarRecentlyClosed = false;
-    }, 300);
+  function toggleFloatingElement() {
+    open = !open;
   }
-
-  $: currentSelection = $dashboardStore?.selectedTimeRange?.name;
-  $: intermediateSelection = currentSelection;
-
-  const handleMenuOpen = () => {
-    if (intermediateSelection !== TimeRangePreset.CUSTOM) {
-      isCustomRangeOpen = false;
-    }
-  };
 </script>
 
-<WithTogglableFloatingElement
```

Comment:
gotta love any PR that removes an instance of `WithTogglableFloatingElement` ;-)